packages/core/src/new-api/types/journal.ts

```diff
@@ -0,0 +1,197 @@
+import { ArgumentType, FutureType } from "./module";
+
+/**
+ * Store a deployments execution state as a transaction log.
+ *
+ * @beta
+ */
+export interface Journal {
+  record(message: JournalableMessage): Promise<void>;
+
+  read(): AsyncGenerator<JournalableMessage>;
+}
+
+/**
+ * A message recordable in the journal's transaction log.
+ *
+ * @beta
+ */
+export type JournalableMessage = TransactionMessage | ExecutionMessage;
+
+// #region "TransactionMessage"
+
+/**
+ * The journal message relating to transaction service state.
+ *
+ * @beta
+ */
+export type TransactionMessage = OnchainInteraction | OnchainResult;
+
+// #region "OnchainInteraction"
+
+/**
+ * A on-chain interaction request for the transaction service.
+ *
+ * @beta
+ */
+export type OnchainInteraction = DeployContractInteraction;
```

Comment:
I think it would be easier to understand if all these have a shared name pattern, like all of them ending in `Message` or `JournalMessage`.

Reply:
I have appended `Message` to each messages name.